wordpress-mobile/WordPress-Android
# Support multiple media sources

Fixes #13089

This PR adds the support for multiple data sources. It is currently used in the featured image selection and in the site icon selection (I don't know if it's necessary somewhere else, I couldn't find any other place where it could be used). I think we'll replace this solution by tabs soon so this should be temporary.

I've added a WP media library as an optional icon, it's visible if it fits the screen. The stock media and gif selection are always collapsed in the menu.

Tenor selection is not yet enabled because it's not yet implemented.

<img width="510" alt="Screenshot 2020-10-08 at 16 30 30" src="https://user-images.githubusercontent.com/1079756/95476584-ab0e2d00-0987-11eb-88b2-2dd3339596a6.png">
<img width="511" alt="Screenshot 2020-10-08 at 16 30 37" src="https://user-images.githubusercontent.com/1079756/95476594-ad708700-0987-11eb-8f15-a527a5e8c90f.png">
<img width="510" alt="Screenshot 2020-10-08 at 16 30 55" src="https://user-images.githubusercontent.com/1079756/95476599-afd2e100-0987-11eb-9469-8aec9e513ad1.png">


To test:
- Turn on the Consolidated media picker flag
- Go to a post editor/settings/set featured image
- Notice that there is now the extra menu in the toolbar
- Notice that it contains 3 options to select from WP Media library, stock library and gif library
- Click on WP Media library
- Notice a new activity is opened 
- Notice the extra menu is now gone
- Select an image and confirm it
- The featured image gets selected

To test:
- Turn on the Consolidated media picker flag
- Go to a post editor/settings/set featured image
- Notice that there is now the extra menu in the toolbar
- Notice that it contains 3 options to select from WP Media library, stock library and gif library
- Click on the Stock library
- Notice a new activity is opened 
- Notice the search is now enabled as default
- Search for an image
- Select an image and confirm it
- The featured image gets selected

To test:
- Turn on the Consolidated media picker flag
- Go to my site/click on site icon/change icon
- Notice that there is now the WP media option available
- Select a site icon
- Check that it's now selected
 

PR submission checklist:
